sfepy/terms/terms_multilinear.py

```diff
@@ -1529,3 +1522,160 @@ def get_function(self, mat, parameter, mode=None, term_mode=None,
+    arg_shapes = {'material' : 'D, D', 'virtual': (1, 'state'), 'state': 1,
+                  'parameter_1': 1, 'parameter_2': 1}
+    modes = ('weak', 'eval')
+
+    def get_function(self, mat, vvar, svar,
+                     mode=None, term_mode=None, diff_var=None, **kwargs):
+
+        fun = self.make_function(
+            'ij,0.i,0.j', mat, vvar, svar, diff_var=diff_var
+        )
+
+        return fun
+
+
+class ELinearTractionTerm(ETermBase):
+    r"""
+    Linear traction term. The material parameter can have one of the
+    following shapes:
+
+        - 1 or (1, 1) - a given scalar pressure
+        - (D, 1) - a traction vector
+        - (S, 1), (2D, 1) or (D, D) - a given stress in symmetric or
+          non-symmetric tensor storage (in symmetric storage indicies are order
+          as follows: 2D: [11, 22, 12], 3D: [11, 22, 33, 12, 13, 23])
+
+    :Definition:
+
+    .. math::
+        \int_{\Gamma} \ul{v} \cdot \ul{n}
+            \mbox{ , }
+        \int_{\Gamma} c\, \ul{v} \cdot \ul{n}\\
+        \int_{\Gamma} \ul{v} \cdot (\ull{\sigma}\, \ul{n})
+            \mbox{ , }
+        \int_{\Gamma} \ul{v} \cdot \ul{f}
+
+    :Arguments:
+        - material: :math:`c`, :math:`\ul{f}`, :math:`\ul{\sigma}`
+          or :math:`\ull{\sigma}`
+        - virtual/parameter: :math:`\ul{v}`
+    """
+    name = 'de_surface_ltr'
+    arg_types = (('opt_material', 'virtual'),
+                 ('opt_material', 'parameter'))
+    arg_shapes = [{'opt_material': 'S, 1', 'virtual': ('D', None),
+                   'parameter': 'D'},
+                  {'opt_material': None}, {'opt_material': '1, 1'},
+                  {'opt_material': 'D, 1'}, {'opt_material': 'D, D'}]
+    modes = ('weak', 'eval')
+    integration = 'surface'
+
+    def get_function(self, traction, vvar, mode=None, term_mode=None,
+                     diff_var=None, **kwargs):
+        sg, _ = self.get_mapping(vvar)
+        nv = sg.normal
+        _, n_qp, dim, _ = nv.shape
+
+        tdim, tdim2 = (None, None) if traction is None else traction.shape[2:]
+ 
+        if tdim == dim and tdim2 == 1:  # force vector
+            force = traction
+        else:
+            sym = (dim + 1) * dim // 2
+
+            if tdim is None:
+                force = nv
+            elif tdim == 1:  # scalar value
+                force = traction * nv
+            elif tdim == dim and tdim2 == dim:  # traction tensor - full
+                force = dot_sequences(traction, nv, 'AB')
+            elif tdim == dim**2 and tdim2 == 1:  # traction tensor - full, vector
+                aux = traction.reshape((-1, n_qp, dim, dim))
+                force = dot_sequences(aux, nv, 'AB')
+            elif tdim == sym and tdim2 == 1:  # traction tensor - symetric
+                aux = sym2nonsym(traction, [2]).reshape((-1, n_qp, dim, dim))
+                force = dot_sequences(aux, nv, 'AB')
+            else:
+                raise NotImplemented
```

Comment:
`NotImplementedError('unsupported material shape! ({})'.format((tdim, tdim2)))` - is not this caught by `arg_shapes`?

Reply:
Yes, can be removed.